packages/grafana-ui/src/components/RefreshPicker/RefreshPicker.tsx

```diff
@@ -72,42 +69,30 @@ export class RefreshPicker extends PureComponent<Props> {
   }
 
   render() {
-    const {
-      onRefresh,
-      intervals,
-      tooltip,
-      value,
-      text,
-      isLoading,
-      noIntervalPicker,
-      width,
-      offOptionLabelMsg,
-      offOptionAriaLabelMsg,
-      offDescriptionAriaLabelMsg,
-      onDescriptionAriaLabelMsg,
-    } = this.props;
+    const { onRefresh, intervals, tooltip, value, text, isLoading, noIntervalPicker, width } = this.props;
 
     const currentValue = value || '';
     const variant = this.getVariant();
-    const translatedOffOption = {
-      value: RefreshPicker.offOption.value,
-      label: offOptionLabelMsg || RefreshPicker.offOption.label,
-      ariaLabel: offOptionAriaLabelMsg || RefreshPicker.offOption.ariaLabel,
-    };
-    const options = intervalsToOptions({ intervals, offOption: translatedOffOption });
+    const options = intervalsToOptions({ intervals });
     const option = options.find(({ value }) => value === currentValue);
+    const translatedOffOption = translateOptions(RefreshPicker.offOption.value);
     let selectedValue = option || translatedOffOption;
 
     if (selectedValue.label === translatedOffOption.label) {
       selectedValue = { value: '' };
     }
 
     const durationAriaLabel = selectedValue.ariaLabel;
-    const ariaLabel =
-      selectedValue.value === ''
-        ? offDescriptionAriaLabelMsg || 'Auto refresh turned off. Choose refresh time interval'
-        : onDescriptionAriaLabelMsg?.(durationAriaLabel) ||
-          `Choose refresh time interval with current interval ${durationAriaLabel} selected`;
+    const ariaLabelDurationSelectedMessage = t(
+      'refresh-picker.aria-label.duration-selected',
+      'Choose refresh time interval with current interval {{durationAriaLabel}} selected',
+      { durationAriaLabel }
+    );
+    const ariaLabelChooseIntervalMessage = t(
+      'refresh-picker.aria-label.choose-interval',
+      'Auto refresh turned off. Choose refresh time interval'
+    );
+    const ariaLabel = selectedValue.value === '' ? ariaLabelChooseIntervalMessage : ariaLabelDurationSelectedMessage;
```

Comment:
@JoaoSilvaGrafana, as this is the same `aria-label` but with a different value, I finally called the i18nKey `refresh-picker.aria-label.-`. What do you think?

Reply:
Yeah I think this is a good idea actually, at first I found it strange to read, but actually with your explanation I think it makes sense 👍